### DISCUSSION TITLE
[EDIT: was actually a bug with very long URLs] Prune history as it gets bigger

### DISCUSSION THREAD

**Discussion by choucavalier** [qutebrowser/qutebrowser#7760] [Q&A] [answered]
When I open a new tab/window, it's often pretty slow and scrambling as I type because it's in real-time looking into my history for matches with what I already typed.

It would be nice if qutebrowser could be configured to prune its history. My thinking is that domains that have not been accessed for e.g. more than a month could be removed or something like that.

**Comment by The-Compiler** [comment, marked as answer]
You can set `completion.web_history.max_items` to only show the N newest items when completing an URL. The entire history is still kept for things like `:history`, as only the completion seems to be affected by this.

That being said, you might also be running into things like #1099. I have 478k entries in my history database without any noticeable issues.

**Reply by choucavalier** [reply]
As always, you are awesome. I had exactly that problem: a URL with 27k+ characters... Seems solved now. Deleted all History entries with len(url)>300 which I think is really a maximum...
